test/integration/targets/win_pester/tasks/main.yml

```diff
@@ -0,0 +1,117 @@
+---
+- name: Get facts
+  setup:
+
+- name: Add Pester module
+  win_psmodule:
+    name: Pester
+    state: present
+  when: ansible_powershell_version >= 5
+
+- name: Add Pester module with Chocolatey
+  win_chocolatey:
+    name: Pester
+    state: present
+  when: ansible_powershell_version < 5
+
+- name: Copy test file(s)
+  win_copy:
+    src: "{{ item }}"
+    dest: "{{ remote_test_folder[0] }}"
+  with_items: "{{local_test_files}}"
+
+- name: Run Pester test(s) specifying a fake test file
+  win_pester:
+    src: "{{ fake_remote_test_file }}"
+  register: pester_result
+  ignore_errors: yes
+
+- name: Assert pester_result
+  assert:
+    that: 
+      - pester_result.failed == true
+
+- name: Run Pester test(s) specifying a fake folder
+  win_pester:
+    src: "{{ fake_remote_folder }}"
+  register: pester_result
+  ignore_errors: yes
+
+- name: Assert pester_result
+  assert:
+    that: 
+      - pester_result.failed == true
+
+- name: Run Pester test(s) specifying a test file and a higher pester version
+  win_pester:
+    src: "{{ remote_test_folder[0] }}{{ remote_test_files[0] }}"
+    version: 6.0.0
+  register: pester_result
+  ignore_errors: yes
+
+- name: Assert pester_result
+  assert:
+    that: 
+      - pester_result.failed == true
+
+- name: Run Pester test(s) specifying a test file
+  win_pester:
+    src: "{{ remote_test_folder[0] }}{{ remote_test_files[0] }}"
+  register: pester_result
+
+- name: Assert pester_result
+  assert:
+    that: 
+      - pester_result.failed == false
+      - pester_result.pester_result.TotalCount == 1
+
+- name: Run Pester test(s) specifying a test file and with a minimum mandatory Pester version
+  win_pester:
+    src: "{{ remote_test_folder[0] }}{{ remote_test_files[0] }}"
+    version: '3.0.0'
```

Comment:
Strictly speaking quoting is not needed here, YAML cannot interpret it as a float.
In this case quoting is useful so people avoid the issue with a single dot.